### ISSUE TITLE
Sending a reaction bolds the room

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#9952]
until I move my mouse again.

**Comment by jryans** [comment]
Hmm, does the bolding happen immediately after sending the reaction? I don't think I have noticed this in testing yet, but perhaps we have different account settings for the room list, etc.?

**Comment by turt2live** [comment]
Immediately after, yes. I have a slow homeserver, which may be contributing to the visibility of the problem.

**Comment by jryans** [comment]
We may need to tweak the unread state processing for reactions (or relations more generally). Our current UX thinking is that relations should not affect the unread state of the room.

**Comment by ara4n** [comment]
i can't reproduce this - @turt2live, pics or it didn't happen?

**Comment by turt2live** [comment]
Will get a gif later today. It's definitely happening.

**Comment by turt2live** [comment]
Note how adding a +1 to a random message bolds the room on the left momentarily. While testing, I noticed that this only happens when there's been no other reactions after the most recent message (meaning I'm the first "hidden" event to be sent).

![reactionbold mp4](https://user-images.githubusercontent.com/1190097/59367060-9587e500-8cf8-11e9-98cc-6ae6f98e5974.gif)

**Comment by jryans** [comment]
So far, I can't seem to reproduce this after trying various accounts, room types, messages, etc. I tried the case of ensuring the reaction is the first hidden event, but that did not seem to be enough to trigger it for me.

@turt2live I think I may need some help isolating the condition to reproduce this. 

* Are you using https://riot.im/develop above, or something with a custom config?
* Do you see this issue with a new test account on your HS (with reactions enabled)?
* Do you see this issue with a new test account on matrix.org HS (with reactions enabled)?
* Does it happen in both clear text and E2E rooms?
* Does it happen in both group and DM rooms?
* Does the message you react to need to be from someone other than yourself?
* Does the message have to have a URL preview?

**Comment by jryans** [comment]
Hmm, well, I did just observe this happening in a DM room with @nadonomy. I'll try to isolate why it happens there.

**Comment by turt2live** [comment]
In answer to your questions:
* Yes, on develop with the labs settings. Using Chrome.
* Yes, tested with t2l.io on riot.im/develop using an alt account.
* Yes, tested with a new matrix.org account
* Yes, both e2e and non-e2e rooms
* Yes, both group and DM rooms
* Yes, it needs to be someone else's message
* No, it does not need a preview

**Comment by jryans** [comment]
This occurs while the reaction is in local echo and clears up once we receive the remote echo.

Possible solutions include one or more of:

* Teach `doesRoomHaveUnreadMessages` about the pending event list and set unread to `false` if there are pending events
* Include reactions in `shouldHideEvent`
* Ban reactions from triggering in `eventTriggersUnreadCount`

(We really need to de-duplicate the N different interconnected functions for show / hide / trigger).

**Comment by jryans** [comment]
I now have more detail about what's going wrong here:

1. We recently added support to local echo reactions
2. We also changed the read receipt behaviour to advance past hidden events like reactions
3. It can happen that your read receipt advances past a pending reaction
4. The unread state partly relies on the read receipt, but (correctly) doesn't expect it to descend into pending events

So, my new approach is to ensure the read receipt cannot advance into pending events.
